contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -1234,6 +1234,41 @@ class tsqlCommonMutator : public TSqlParserBaseListener
 		if (!(in_create_or_alter_function || in_create_or_alter_procedure || in_create_or_alter_trigger))
 			rewrite_string_agg_query(ctx);
 	}
+
+	/*
+	 * The $PARTITION function handling is implemented in the
+	 * tsqlCommonMutator to support its usage not only from direct calls
+	 * but also within Views, Procedures, and Functions.
+	 */
+	void exitPartition_function_call(TSqlParser::Partition_function_callContext *ctx) override
+	{
+		/*
+		 * Re-write db_name.$PARTITION.partition_func_name(exp)
+		 * to sys.search_partition(partition_func_name, exp, db_name)
+		 * where db_name is optional.
+		 */
+
+		/* Remove dot after the $PARTITION and partition function name from query. */
+		rewritten_query_fragment.emplace(std::make_pair(ctx->DOT().back()->getSymbol()->getStartIndex(), std::make_pair(::getFullText(ctx->DOT().back()), "")));
+		rewritten_query_fragment.emplace(std::make_pair(ctx->id().back()->start->getStartIndex(), std::make_pair(::getFullText(ctx->id().back()), "")));
+		
+		/* Replace "$PARTITION" with "sys.search_partition". */
+		rewritten_query_fragment.emplace(std::make_pair(ctx->DOLLAR_PARTITION()->getSymbol()->getStartIndex(), std::make_pair(::getFullText(ctx->DOLLAR_PARTITION()),
```

Comment:
It would be better that we add a boolean to the parse tree(partition stmt already added) for dollar_partition and set it true along with the partition_func_name, exp, db_name and then later in pl_exec we call C function directly instead of this sql one